sdk/ai/azure-ai-assistants/tests/model_assistants_test_base.py

```diff
@@ -0,0 +1,493 @@
+# AZURE_AI_CHAT_KEY=<32-char-api-key>
+#
+ServicePreparerChatCompletions = functools.partial(
+    EnvironmentVariableLoader,
+    "azure_ai_chat",
+    azure_ai_chat_endpoint="https://your-deployment-name.eastus2.inference.ai.azure.com",
+    azure_ai_chat_key="00000000000000000000000000000000",
+)
+
+#
+# Define these environment variables. They should point to any GPT model that
+# accepts image input in chat completions (e.g. GPT-4o model).
+# hosted on Azure OpenAI (AOAI) endpoint.
+# TODO: When we have a MaaS model that supports chat completions with image input,
+# use that instead.
+# AZURE_OPENAI_CHAT_ENDPOINT=https://<endpont-name>.openai.azure.com/openai/deployments/gpt-4o
+# AZURE_OPENAI_CHAT_KEY=<32-char-api-key>
+#
+ServicePreparerAOAIChatCompletions = functools.partial(
+    EnvironmentVariableLoader,
+    "azure_openai_chat",
+    azure_openai_chat_endpoint="https://your-deployment-name.openai.azure.com/openai/deployments/gpt-4o",
+    azure_openai_chat_key="00000000000000000000000000000000",
+)
+
+#
+# Define these environment variables. They should point to a Cohere model
+# hosted on MaaS, or any other MaaS model that text embeddings.
+# AZURE_AI_EMBEDDINGS_ENDPOINT=https://<endpoint-name>.<azure-region>.models.ai.azure.com
+# AZURE_AI_EMBEDDINGS_KEY=<32-char-api-key>
+#
+ServicePreparerAssistantClient = functools.partial(
+    EnvironmentVariableLoader,
+    "azure_ai_assistant_client",
+    azure_ai_assistant_client_endpoint="https://your-deployment-name.openai.azure.com/", 
+    azure_ai_assistant_client_key="00000000000000000000000000000000", 
+    azure_ai_assistant_client_api_version=os.environ.get("AZUREAI_API_VERSION", "2024-07-01-preview")
+)
+
+
+# The test class name needs to start with "Test" to get collected by pytest
+class ModelClientTestBase(AzureRecordedTestCase):
+
+    # Set to True to print out all results to the console
+    PRINT_RESULT = True
+
+    # Regular expression describing the pattern of a result ID returned from MaaS/MaaP endpoint. Format allowed are:
+    # "183b56eb-8512-484d-be50-5d8df82301a2", "26ef25aa45424781865a2d38a4484274" and "Sanitized" (when running tests
+    # from recordings)
+    REGEX_RESULT_ID = re.compile(
+        r"^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$|^[0-9a-fA-F]{32}$|^Sanitized$"
+    )
+
+    # Regular expression describing the pattern of a result ID returned from AOAI endpoint.
+    # For example: "chatcmpl-9jscXwejvOMnGrxRfACmNrCCdiwWb" or "Sanitized" (when runing tests from recordings) # cspell:disable-line
+    REGEX_AOAI_RESULT_ID = re.compile(r"^chatcmpl-[0-9a-zA-Z]{29}$|^Sanitized$")  # cspell:disable-line
+
+    # Regular expression describing the pattern of a base64 string
+    REGEX_BASE64_STRING = re.compile(r"^[A-Za-z0-9+/]+={0,3}$")
+
+
+    # TODO: sophia!!!! figure out how to not need chatcompletionstool since assistant doesnt have that!  

```

Comment:
Tools for assistants have currently different abstraction than chat completion tools